aux_process.c

```diff
@@ -0,0 +1,1537 @@
+    uint64_t cl_tick_num;
+
+    uint32_t md_file_header_ud_page_offset; 
+    uint32_t md_file_header_len;
+    uint32_t md_file_header_chksum;
+    void     *md_file_header_buf;
+
+    uint32_t md_file_index_ud_page_offset;
+    uint64_t md_file_index_md_file_offset;
+    uint32_t md_file_index_len;
+    uint32_t md_file_index_chksum;
+    void     *md_file_index_buf;
+
+    uint32_t received_cl_checksum;
+    uint32_t verified_cl_checksum;
+
+    uint32_t num_cl_entries;
+    cl_entry_t *change_list;
+    uint32_t cl_chksum; 
+} updater_t;
+
+enum aux_arg_level {
+    no_arg = 0,  /* doesn't take an argument     */
+    require_arg, /* requires an argument          */
+    optional_arg /* argument is optional         */
+};
+
+/*
+ * aux_get_options is a copy of the H5_get_options in hdf5/src/H5system.c.
+ *
+ * It supports both POSIX and Windows systems.
+ * It determines which options are specified on the command line and
+ * returns a pointer to any arguments possibly associated with the option in
+ * the ``aux_optarg'' variable. aux_get_options returns the shortname equivalent of
+ * the option. The long options are specified in the following way:
+ *
+ * struct aux_long_options foo[] = {
+ *   { "filename", require_arg, 'f' },
+ *   { "append", no_arg, 'a' },
+ *   { "width", require_arg, 'w' },
+ *   { NULL, 0, 0 }
+ * };
+ *
+ * Long named options can have arguments specified as either:
+ *
+ *   ``--param=arg'' or ``--param arg''
+ *
+ * Short named options can have arguments specified as either:
+ *
+ *   ``-w80'' or ``-w 80''
+ *
+ * and can have more than one short named option specified at one time:
+ *
+ *   -aw80
+ *
+ * in which case those options which expect an argument need to come at the
+ * end.
+ */
+typedef struct {
+    const char *      name;     /* Name of the long option */
+    enum aux_arg_level has_arg;  /* Whether we should look for an arg */
+    char              shortval; /* The shortname equivalent of long arg
+                                 * this gets returned from get_option
+                                 */
+} aux_long_options;
+
+
+int         aux_opterr = 1; /* Get_option prints errors if this is on */
+int         aux_optind = 1; /* Token pointer                          */
+const char *aux_optarg;     /* Flag argument (or value)               */
+
+/*-------------------------------------------------------------------------
+ * Function: aux_get_options
+ *
+ * Purpose:  Determine the command-line options a user specified. We can
+ *           accept both short and long type command-lines.
+ *
+ * Return:  Success:    The short valued "name" of the command line
+ *                      parameter or EOF if there are no more
+ *                      parameters to process.
+ *
+ *          Failure:    A question mark.
+ *-------------------------------------------------------------------------
+ */
+int
+aux_get_options(int argc, char **argv, const char *opts, const aux_long_options *l_opts)
```

Comment:
Why recreate options handling - getopt and such are in the library - see other tools.

Reply:
This function handles Windows.  `getopt` is a POSIX function.  Originally, we wanted this program to be totally independent of the library.  That's why I copied it from the library.  Later, it became a submodule of the HDF5 library.  I need to change it to use the library's own function instead of copying it here.

Reply:
Right, the options functions used to be in the tools library, then moved into the library. I think any tools built to link with the hdf5 library should use the functions in the library.

Reply:
I told the team about this function.  The consensus was to keep the way it is now because this submodule may get fulled out in the future.